src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -475,5 +470,72 @@ private static bool NotNull(SymbolReference reference)
         {
             return reference.SymbolResult.Symbol != null;
         }
+
+        public async Task<ImmutableArray<(Diagnostic Diagnostic, ImmutableArray<AddImportFixData> Fixes)>> GetFixesForDiagnosticsAsync(
+            Document document, TextSpan span, ImmutableArray<Diagnostic> diagnostics,
+            ISymbolSearchService symbolSearchService, bool searchReferenceAssemblies, 
+            ImmutableArray<PackageSource> packageSources, CancellationToken cancellationToken)
+        {
+            // We might have multiple different diagnostics covering the same span.  Have to
+            // process them all as we might produce different fixes for each diagnostic.
+
+            var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var placeSystemNamespaceFirst = documentOptions.GetOption(GenerationOptions.PlaceSystemNamespaceFirst);
+
+            var fixesForDiagnosticBuilder = ArrayBuilder<(Diagnostic, ImmutableArray<AddImportFixData>)>.GetInstance();
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var fixes = await GetFixesAsync(
+                    document, span, diagnostic.Id, placeSystemNamespaceFirst,
+                    symbolSearchService, searchReferenceAssemblies,
+                    packageSources, cancellationToken).ConfigureAwait(false);
+
+                fixesForDiagnosticBuilder.Add((diagnostic, fixes));
+            }
+
+            return fixesForDiagnosticBuilder.ToImmutableAndFree();
+        }
+
+        public ImmutableArray<CodeAction> GetCodeActionsForFixes(Document document, ImmutableArray<AddImportFixData> fixes, IPackageInstallerService installerService, bool limitResults)
```

Comment:
plz wrap super long parameter lists :)

good rule of thumb: if it's too wide for github, it's too wide.